### ISSUE TITLE
Feature request: tags support

### ISSUE THREAD

**Issue by igerasimov23** [BorisOsipov/wdio-reportportal-reporter#19]
## The problem

Feature request:
Report portal API supports adding tags to the test. Can we extend functionality of wdio-reportportal-reporter

## Environment

* WebdriverIO version: 4.11
* wdio-reportportal-version version: 0.0.13
* Node.js version: v8.10
* WDIO Testrunner and version(mocha\cucumber\jasmine):
* Additional wdio packages used (if applicable):

**Comment by BorisOsipov** [comment]
@igerasimov23 Please add code snippet how you handle tags in your code.

**Comment by igerasimov23** [comment]
As mocha does not support tagging like in JAVA annotations, I was thinking just parse test case name for tagging and just add it to `testStartObj` in testStart function in reporter.js.
My idea was as an example:
parse test case name `Verify footer is displayed -tags tag1, tag2, tag3` and pass `tag1, tag2, tag3` to 
 const testStartObj = { type: entityType.TEST, name: test.title, tags: ['tag1','tag2', 'tag3'] };

OR 
test case name  `Verify footer is displayed @tag tag1  @tag tag2  @tag tag3`

I looked on reporter.js in wdio-reportportal-reporter and it's very easy should be implement

What do you think will be better approach?

**Comment by igerasimov23** [comment]
For my approach we can just add to https://github.com/BorisOsipov/wdio-reportportal-reporter/blob/master/lib/reporter.js#L117 :
```
let tags = test.title;
    let tagsArray = [];
    if(tags.indexOf('@tags') !== -1) {
      tags  = tags.substr(tags.indexOf('@tags') + 6);
      if (tags.indexOf(' ') >= 0)
        tags = tags.substr(0, tags.indexOf(' '));
      tagsArray = tags.split(',');
    }

    const testStartObj = { type: entityType.TEST, name: test.title, tags: tagsArray };
```
Any thoughts?

**Comment by BorisOsipov** [comment]
Yes I get what you want to achive and how. I think parsing test title is ok.
Some thoughts:
1) We should declare some strong format\rules for tags parsing e.g. `@tag1(,|;)tag2 = [tag1, tag2]`, `@tag1,@tag2 = [tag1, tag2]`. Or maybe require to always start each tag with `@` e.g. `@tag1 some text @tag2 = [tag1, tag2]`. It almost up to you, just think how it will be easy to use and reliable to parse. My main concern - I don't want to change this logic a lot in the future because in will require significant(?) changes in your tests. So it requires careful thinking how to do it.
2) It should be enabled\disabled by feature flag cuz someone may use `@` in test title directly e.g. in case of email in the test title: `Should fail with wrong email foo@bar`
Do you agree?
3) I wouldn't like see such logic directly in `testStart` - need extract to some utils method
4) Also some unit tests for such utils method must be.

**Comment by BorisOsipov** [comment]
@igerasimov23 What do you think about removing tags from test title for RP name?
E.g. test
`test.title = 'Verify footer is displayed @Smoke @Footer @Foo'`  
will send to RP as  
`testStartObj = {name = 'Verify footer is displayed', tags= ['@Smoke', '@Footer', '@Foo']`.
